2.13.3/OmiseExt/class/models/Omise_Models_Charge.php

```diff
@@ -0,0 +1,271 @@
+<?php
+/**
+ * This file is part of EC-CUBE Plugin OmiseExt
+ *
+ * @copyright 2016 Omise All Rights Reserved.
+ * @author Akira Narita
+ *
+ */
+
+/**
+ * Omise charge handler
+ *
+ * @author Akira / Omise
+ */
+class Omise_Models_Charge
+{
+    /**
+     * @var $arrOrder テーブルからロードした order のデータ
+     */
+    public $arrOrder;
+
+    public function __construct($order_id)
+    {
+        $objPurchase    = new SC_Helper_Purchase_Ex();
+        $this->arrOrder = $objPurchase->getOrder($order_id);
+    }
+
+    /**
+     * Sync Omise Charge data and local data on ECCUBE db
+     */
+    public function syncOmise()
+    {
+        $objCharge   = OmiseWrapper::chargeRetrieve(self::getChargeId());
+        $objPurchase = new SC_Helper_Purchase_Ex();
+        $updateData  = array(OMISE_MDL_CHARGE_DATA_COL => $this->lfConvertToDbChargeData($objCharge));
+        $objQuery    = SC_Query_Ex::getSingletonInstance();
+        $objQuery->begin();
+        $objPurchase->sfUpdateOrderStatus(
+            $this->arrOrder['order_id'],
+            null, // 対応状況 ＊nullは変更なし
+            null, // 加算ポイント ＊nullは変更なし
+            null, // 使用ポイント ＊nullは変更なし
+            $updateData
+        );
+        $objQuery->commit();
+        $order_id = $this->arrOrder['order_id'];
+        $this->arrOrder = $objPurchase->getOrder($order_id);
+    }
+
+    /**
+     * Get Omise Charge id
+     * @return string Omise Charge id string
+     */
+    public function getChargeId()
+    {
+        $arrData = $this->lfGetChargeData();
+        return $arrData['id'];
+    }
+
+    /**
+     * Return true if refunded
+     * @return boolean
+     */
+    public function isRefunded()
+    {
+        $arrData = $this->lfGetChargeData();
+        return ($arrData['real_amount'] == 0) ? true : false;
+    }
+
+    /**
+     * Return true if livemode
+     * @return boolean
+     */
+    public function isLiveCharge()
+    {
+        $arrData = $this->lfGetChargeData();
+        return $arrData['livemode'];
+    }
+
+    /**
+     * Get the Omise Charge detail page url string
+     * @return string The Omise Charge detail page
+     */
+    public function getOmisePage()
+    {
+        return sprintf('https://dashboard.omise.co/%s/charges/%s',
+            $this->isLiveCharge() ? 'live' : 'test',
+            $this->getChargeId());
+    }
+
+    /**
+     * Return true if the Omise Charge was captured
+     * @return boolean
+     */
+    public function isCaptured()
+    {
+        $arrData = $this->lfGetChargeData();
+        return ($arrData['captured'] == 1) ? true : false;
```

Comment:
@akinrt since Omise API version 2015-xx-xx returns `paid` instead of `captured`. 
 You have to bypass API version to 2014-xx-xx instead if you want to check this `captured` param

You can put `define('OMISE_API_VERSION', '2014-07-27');` to somewhere before your createCharge method to bypass the version.